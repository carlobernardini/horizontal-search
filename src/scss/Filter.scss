@import 'base';

.orizzonte__filter {
	background-color: $filter-color-idle;
	border-radius: $filter-radius;
	transition: all .3s;
	white-space: nowrap;
	@extend %btn;

	&:hover {
		background-color: $filter-color-active;

		.orizzonte__filter-btn {
			width: 20px;
			transform: scale(1);
			opacity: 1;
		}
	}

	+ .orizzonte__filter {
		margin-left: 5px;
	}

	&--shown {
		background-color: lighten($filter-color-active, 10%);
		box-shadow: 0 2px 4px rgba(0, 0, 0, .2);

		.orizzonte__filter-btn {
			width: 20px;
			transform: scale(1);
			opacity: 1;
		}

		&:hover {
			background-color: lighten($filter-color-active, 10%);
		}
	}

	&--removing {
		overflow: hidden;
		animation: elementOut .3s forwards;
	}
}

.orizzonte__filter-label {
	margin: -7px 0 -7px -15px;
	padding: 7px 0 7px 15px;
	background-color: transparent;
	outline: none;
	border: none;
	cursor: pointer;
	@extend %btn-label;
}

.orizzonte__filter-btn {
	overflow: hidden;
	margin: {
		left: 5px;
		right: -8px;
		bottom: -7px;
	}
	position: relative;
	top: -1px;
	padding: 0;
	width: 0px;
	line-height: 20px;
	text-align: center;
	color: #FFF;
	border: none;
	outline: none;
	color: $filter-color-active;
	background-color: rgba(255, 255, 255, 1);
	border-radius: 25px;
	opacity: 0;
	transform: scale(0);
	cursor: pointer;
	transition: all .3s;

	&:active {
		transform: scale(1.5);
	}
}
