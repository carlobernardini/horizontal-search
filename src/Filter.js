import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './scss/Filter.scss';

class Filter extends Component {
	constructor(props) {
        super(props);

        this.state = {
        	shown: false,
        	removing: false
        };

        this.removeFilter = this.removeFilter.bind(this);
    }

    removeFilter(e) {
    	const { i, onFilterRemove } = this.props;

    	this.setState({
    		shown: false,
    		removing: true
    	});

    	setTimeout(onFilterRemove.bind(null, i), 300);
    }

	render() {
		const { name, onFilterRemove } = this.props;
		const { shown, removing } = this.state;

		return (<div
			className={ classNames('orizzonte__filter', {
				'orizzonte__filter--shown': shown,
				'orizzonte__filter--removing': removing
			}) }
		>
			<a
				href="#"
				onClick={ (e) => {
					e.preventDefault();
					this.setState({
						shown: !shown
					});
				}}
				className="orizzonte__filter-label"
				
			>
				{ name }
			</a>
			<button
				type="button"
				className="orizzonte__filter-btn"
				onClick={ this.removeFilter }
			>
				&times;
			</button>
		</div>);
	}
}

Filter.propTypes = {
	name: PropTypes.string.isRequired,
	onFilterRemove: PropTypes.func
};

Filter.defaultProps = {
	onFilterRemove: () => {}
};

export default Filter;
