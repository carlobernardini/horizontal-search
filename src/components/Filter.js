import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import '../scss/Filter.scss';

class Filter extends Component {
    constructor(props) {
        super(props);

        this.state = {
            shown: false,
            removing: false
        };

        this.removeFilter = this.removeFilter.bind(this);
        this.toggleFilter = this.toggleFilter.bind(this);
    }

    removeFilter() {
        const { i, onFilterRemove } = this.props;

        this.setState({
            shown: false,
            removing: true
        });

        setTimeout(onFilterRemove.bind(null, i), 300);
    }

    toggleFilter() {
        const { shown } = this.state;

        this.setState({
            shown: !shown
        });
    }

    renderBtn() {
        const { hideRemove } = this.props;

        if (hideRemove) {
            return null;
        }

        return (
            <button
                type="button"
                className="orizzonte__filter-btn"
                onClick={ this.removeFilter }
            >
                &times;
            </button>
        );
    }

    render() {
        const { label, selected } = this.props;
        const { shown, removing } = this.state;

        if (!selected) {
            return null;
        }

        return (
            <div
                className={ classNames('orizzonte__filter', {
                    'orizzonte__filter--shown': shown,
                    'orizzonte__filter--removing': removing
                }) }
            >
                <button
                    type="button"
                    onClick={ this.toggleFilter }
                    className="orizzonte__filter-label"
                >
                    { label }
                </button>
                { this.renderBtn() }
            </div>
        );
    }
}

Filter.propTypes = {
    /** If a remove button should be present */
    hideRemove: PropTypes.bool,
    /** Internal filter list index */
    i: PropTypes.number,
    /** Filter label */
    label: PropTypes.string.isRequired,
    /** Internal callback for filter removal */
    onFilterRemove: PropTypes.func,
    /** If the filter is added to the bar */
    selected: PropTypes.bool
};

Filter.defaultProps = {
    hideRemove: false,
    i: null,
    onFilterRemove: () => {},
    selected: false
};

export default Filter;
