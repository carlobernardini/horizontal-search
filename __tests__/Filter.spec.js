import React from 'react';
import Filter from '../src/components/Filter';

describe('<Filter />', () => {
    it('should render a simple filter', () => {
        const wrapper = shallow(
            <Filter
                label="Language"
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
