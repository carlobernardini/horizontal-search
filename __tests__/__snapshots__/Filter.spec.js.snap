// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> should render a simple filter 1`] = `
<div
  className="orizzonte__filter"
>
  <button
    className="orizzonte__filter-label"
    onClick={[Function]}
    type="button"
  >
    Language
  </button>
  <button
    className="orizzonte__filter-btn"
    onClick={[Function]}
    type="button"
  >
    Ã—
  </button>
</div>
`;
